### PR TITLE
Fix sass import deprecation

### LINKED ISSUE #24198: Breaking Change, Sass @import rules are deprecated and will be removed in Dart Sass 3.0.0.
### Describe the Bug

Getting @import rules deprecation warnings when running in development mode:

_Deprecation Warning: Sass @import rules are deprecated and will be removed in Dart Sass 3.0.0._

Need to migrate from @import rules to avoid break changes starting with Dart Sass 3.0.0.

Refer to [Breaking Change: `@import` and global built-in functions](https://sass-lang.com/documentation/breaking-changes/import/)

An automatic migration package was created to assist on fixing, that really helps, refer to:
[sass-migrator Reference from breaking-changes](https://sass-lang.com/documentation/breaking-changes/import/#automatic-migration)
[sass-migrator at Github](https://github.com/sass/migrator#readme)
[sass-migrator at npm](https://www.npmjs.com/package/sass-migrator)
[sass-migrator Documentation](https://sass-lang.com/documentation/cli/migrator/)

### To Reproduce

Running Directus in development mode, access the admin and check the container logs, it will output one warning for each @import, as below:

`[20:59:53.021] INFO: Server started at http://0.0.0.0:8055
Deprecation Warning: Sass @import rules are deprecated and will be removed in Dart Sass 3.0.0.

More info and automated migrator: https://sass-lang.com/d/import

  ╷
1 │ @import 'base';
  │         ^^^^^^
  ╵
    /directus/app/src/styles/main.scss 1:9  root stylesheet`

### Directus Version

v11.3.2

### Hosting Strategy

Self-Hosted (Docker Image)

### Database

PostgreSQL 15.6 (not applicable)

### PR DESCRIPTION
**Pull request by robsoncombr**
## Scope

I was able to build it without any error, I have tested and I was not able to find any problem, the deprecation warning is also gone. This was really challenging, I'm not sure I did everything correctly and I don't think I can certify the standard from here on my own, etc. I would appreciate if you could help me. Thanks.

### Find the success build below:

```
* Executing task in folder ubuntu: docker exec -it 4fca7d354821b5a1dda9fb2c7a30c084d75d62c8d3ec595ccb204bd68f682d5d bash

I have no name!@4fca7d354821:/directus$ pnpm --filter app build

@directus/app@13.3.6 build /directus/app
vite build

vite v5.4.10 building for production...
✓ 4422 modules transformed.
[plugin:vite:reporter] [plugin vite:reporter]
(!) /directus/node_modules/.pnpm/date-fns@4.1.0/node_modules/date-fns/locale/en-US.js is dynamically imported by /directus/app/src/utils/get-date-fns-locale.ts but also statically imported by /directus/node_modules/.pnpm/date-fns@4.1.0/node_modules/date-fns/_lib/defaultLocale.js, dynamic import will not move module into another chunk.

[plugin:vite:reporter] [plugin vite:reporter]
(!) /directus/app/src/lang/translations/en-US.yaml is dynamically imported by /directus/app/src/lang/set-language.ts but also statically imported by /directus/app/src/lang/index.ts, dynamic import will not move module into another chunk.

[plugin:vite:reporter] [plugin vite:reporter]
(!) /directus/node_modules/.pnpm/codemirror@5.65.18/node_modules/codemirror/mode/markdown/markdown.js is dynamically imported by /directus/app/src/interfaces/input-code/import-codemirror-mode.ts but also statically imported by /directus/node_modules/.pnpm/codemirror@5.65.18/node_modules/codemirror/mode/markdown/markdown.js?commonjs-proxy, /directus/app/src/interfaces/input-rich-text-md/input-rich-text-md.vue?vue&type=script&setup=true&lang.ts, dynamic import will not move module into another chunk.

[plugin:vite:reporter] [plugin vite:reporter]
(!) /directus/node_modules/.pnpm/codemirror@5.65.18/node_modules/codemirror/mode/javascript/javascript.js is dynamically imported by /directus/app/src/interfaces/input-code/import-codemirror-mode.ts, /directus/app/src/interfaces/input-code/input-code.vue?vue&type=script&setup=true&lang.ts but also statically imported by /directus/node_modules/.pnpm/codemirror@5.65.18/node_modules/codemirror/mode/javascript/javascript.js?commonjs-proxy, /directus/app/src/modules/settings/routes/system-logs/logs.vue?vue&type=script&setup=true&lang.ts, dynamic import will not move module into another chunk.

dist/index.html 1.83 kB │ gzip: 0.74 kB
dist/assets/FiraMono-Medium-CE-6Frqh.woff2 16.15 kB
(.removed.)
dist/assets/index.yA6csgyz.entry.js 6,013.85 kB │ gzip: 1,755.80 kB

(!) Some chunks are larger than 500 kB after minification. Consider:

Using dynamic import() to code-split the application
Use build.rollupOptions.output.manualChunks to improve chunking: https://rollupjs.org/configuration-options/#output-manualchunks
Adjust chunk size limit for this warning via build.chunkSizeWarningLimit.
✓ built in 1m 10s
```

## Potential Risks / Drawbacks

- With this change in place new scss as file or inside vue compoenents should follow new `@use` pattern
- **I consider that it is indispensable that a Senior Front-End review this**, there is a potential chance I did something wrong or not good enough, all I have for now is the building with no error and some fast running tests in browser.

## Review Notes / Questions

- With this change in place new scss as file or inside vue compoenents should follow new `@use` pattern
- I think that because of the pattern change we should have a Changeset bumped on the app, I would also like to have assistance on this matter, thanks.

---

Fixes #24198

### CODE REVIEW

**Comment by rijkvanzanten** [comment]
Holy smokes friend, you've been busy! I saw the errors when @paescuj did his Update Vite PR #23845 and even tried migrating some parts, but gave up as there were a lot of edge cases and it isn't deprecated _yet_ 🫠 Thanks for this!

**Comment by robsoncombr** [comment]
@rijkvanzanten 
_> Holy smokes friend, you've been busy! I saw the errors when @paescuj did his Update Vite PR #23845 and even tried migrating some parts, but gave up as there were a lot of edge cases and it isn't deprecated _yet_ 🫠 Thanks for this!_

I’m very happy to help. When I found Directus, very recently, I simply found the project I'm designing for the last 20 years, but never managing to finish. Three years ago, I wrote some cool code in my newest implementation try and my vision has always been very close to what Directus offers today, have some other ideas that I can share some day.

Also, while I still have your eyes, just to mention, I'm still new and arriving, but looking around, today I noticed discussions about noSQL / MongoDB support implementations #3156. So I wanted to mention that during this last 'self-Directus' implementation try I mentioned, I was using MongoDB and I created 2 nice solutions that I can share with you guys when it is time to work on this MongoDB matter.

- One is a simple workaround for the lack of native full-search feature, that can certainly be improved but might be useful as a beginning or at least for some particular collections, since it does not depend on external tools.
- The other, a microservice I wrote in Node/Express, this piece was very nice, that based on the oplog allow us to implement automations, under the roadmap I had: data streaming, realtime, audit records, backups, revisions, triggers. I implemented a safe-resume to avoid losing any transaction if service is restored from any down time event.

Who knows, maybe some time I create a repository to open-source it, for sure might be useful for someone.

Well, gotta go for today, I hope I’ve been helpful, if you have any questions, just let me know!

# Fly high, Directus! 🚀

**Review by connorwinston — DISMISSED** [review]
Wow!! Thanks for this PR @robsoncombr. From my testing this all works, just a few things I left comments on.

Since `@use` now imports stuff under a namespace, lets make a index file in `@/style/mixins/` that uses `@forward` to forward all the mixins in that folder under a `mixin` namespace. So that way we can use it like the below.

```scss
@use '@/styles/mixins';

.example-class {
	@include mixins.form-grid;
}
```

Example of `@/styles/mixins/index.scss`
```scss
@forward "form-grid";
@forward "markdown";
@forward "no-wrap";
@forward "type-styles";
```
Let me know if you have any questions!

**Comment by AlexGaillard** [comment]
Thanks for this @robsoncombr 🚀  We're hoping to get this into the next release. So we're going to hand it over internally for someone to finish off if you haven't started with @connorwinston's feedback yet?

**Comment by robsoncombr** [comment]
> Thanks for this @robsoncombr 🚀 We're hoping to get this into the next release. So we're going to hand it over internally for someone to finish off if you haven't started with @connorwinston's feedback yet? 😄

Hey, I'm sorry, I got a little busy and lost this conversation.. I will review now.

File: app/src/styles/themes/mixins/generate-colors.scss [hunk 1]
```diff
@@ -1,22 +1,21 @@
...
-			// Set suffix if not equal to none
+
 			@if $suffix != 'none' {
 				$resolved-suffix: '-#{$suffix}';
 			}
-			// If our step is negative, change to dark mixing color
 			@if $step < 0 {
 				$mixing-color: $mix-color-high;
 			}
-			// This is the CSS variable that will be output
```
[the remark this answers is not in the document]
Comment (robsoncombr) [reply app/src/styles/themes/mixins/generate-colors.scss:18]: Nope, no reason from my part, it was remove by my local pre-commit processor, I will disable and revert, thanks for pointing it out!

File: app/src/styles/main.scss [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
-@import 'lib/highlight';
-@import 'lib/fullcalendar';
-@import 'lib/frappe-charts';
-@import 'lib/emoji-picker';
-@import 'lib/flatpickr';
+@use './base';
+@use './colors' as colors;
+@use './fonts';
+@use './themes/dark' as dark;
+@use './themes/light' as light;
```
[the remark this answers is not in the document]
Comment (robsoncombr) [reply app/src/styles/main.scss:5]: Gotcha, thanks for the guidance! 😎

File: app/src/styles/themes/_dark.scss [hunk 3]
```diff
@@ -27,7 +30,7 @@
 	--sidebar-shadow: var(--navigation-shadow);
 
 	// Generate color variations
-	@include generate-colors($color-mapping, $dark-theme-shade, $dark-theme-tint);
+	@include generate-colors.generate-colors($color-mapping, colors.$dark-theme-tint, colors.$dark-theme-shade);
```
[the remark this answers is not in the document]
Comment (robsoncombr) [reply app/src/styles/themes/_dark.scss:33]: I had a look into my file changes history, I don´t remember to manually change this, good caught, my only theory here is that this was changed by sass-migrator tool, that after experience with this PR, I can say it does not work well, no way to migrate without a lot of manual interventions. Anyway, I will revert it, I'm also a fan of not applying changes that aren't directly related to the task. 😎

**Comment by robsoncombr** [comment]
Ok mates, sorry for the delay, I have reviwed all requests and we are good now.
Let me know any further observation you may have!
Thanks for backing me up. 🤘

**Comment by formfcw** [comment]
@robsoncombr Thank you :rocket: 

I added commits to resolve Connor's request from above https://github.com/directus/directus/pull/24201#pullrequestreview-2510279649 and did some cleanup.

@AlexGaillard Maybe someone can approve only my commits, since the rest LGTM  🎉

**Review by paescuj — APPROVED** [review]
Awesome work, thank you all ❤️

**Pull request merged**